contracts/test/gas.ts

```diff
@@ -0,0 +1,39 @@
+// SPDX-License-Identifier: Apache-2.0
+
+import { expect } from 'chai';
+import { ethers } from 'hardhat';
+import { GasTests__factory } from '../typechain-types/factories/contracts/tests/Gas.sol';
+import { GasTests } from '../typechain-types/contracts/tests/Gas.sol/GasTests';
+
+describe('Gas Padding', function () {
+  let contract: GasTests;
+
+  before(async () => {
+    const factory = (await ethers.getContractFactory(
+      'GasTests',
+    )) as GasTests__factory;
+    contract = await factory.deploy();
+  });
+
+  it('Gas Padding works as Expected', async () => {
+    const expectedGas = 122735;
+
+    let tx = await contract.testConstantTime(1, 100000);
+    let receipt = await tx.wait();
+    expect(receipt.cumulativeGasUsed).eq(expectedGas);
+
+    tx = await contract.testConstantTime(2, 100000);
+    receipt = await tx.wait();
+    expect(receipt.cumulativeGasUsed).eq(expectedGas);
+
+    tx = await contract.testConstantTime(1, 100000);
```

Comment:
This asserts that we are not caching `receipt.cumulativeGasUsed` right?

Reply:
It's a transaction receipt, so the receipt details will be unique to that specific tx.